R/generate.R

```diff
@@ -73,22 +76,41 @@ generate <- function(x, reps = 1, type = NULL, ...) {
       check_permutation_attributes(x)
       permute(x, reps, ...)
     },
-    simulate = simulate(x, reps, ...)
+    draw = draw(x, reps, ...),
+    simulate = draw(x, reps, ...)
   )
 }
 
 # Check that type argument is an implemented type
 sanitize_generation_type <- function(x) {
   if (is.null(x)) return(x)
-  match.arg(x, GENERATION_TYPES)
+  
+  check_type(x, is.character)
+
+  if (!x %in% c("bootstrap", "permute", "simulate", "draw")) {
+    stop_glue(
+      'The `type` argument should be one of "bootstrap", "permute", ',
+      'or "draw". See `?calculate` for more details.'
```

Comment:
Should this say "see `?generate` for more details"?

Reply:
Ah, yes! Thank you.